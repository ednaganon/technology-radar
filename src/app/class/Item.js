export default class Item {
    constructor({ name = '', description = '', category = '', status = '', url = '', tags = '' }) {
        this.name = name;
        this.description = description;
        this.category = category;
        this.status = status;
        this.tags = tags.split(',');
        this.url = url;
    }
}
