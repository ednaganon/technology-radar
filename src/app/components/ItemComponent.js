import React, { Component } from 'react';
import Item from '../class/Item';

export default class ItemComponent extends Component {

    static get propTypes() {
        return {
            item: React.PropTypes.instanceOf(Item),
        };
    }

    render() {
        return (
            <a className="uk-panel uk-panel-box uk-panel-box-default" target="_blank" rel="noopener noreferrer" href={this.props.item.url}>
                <div className="uk-panel-badge uk-badge">{this.props.item.status}</div>
                <h3 className="uk-panel-title">{this.props.item.name}</h3>
                <p>{this.props.item.description}</p>
            </a>
        );
    }

}
