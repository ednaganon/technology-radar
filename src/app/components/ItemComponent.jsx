import React, { Component } from 'react';
import _ from 'underscore';
import Item from '../class/Item';

export default class ItemComponent extends Component {

    static get propTypes() {
        return {
            item: React.PropTypes.instanceOf(Item),
        };
    }

    tags() {
        return _.map(this.props.item.tags, (tag, key) => (
            <span key={key}><div className="uk-label">{tag}</div>&nbsp;</span>
        ));
    }

    render() {
        return (
            <div>
                <a className="uk-card uk-card-default uk-card-hover" target="_blank" rel="noopener noreferrer" {...this.props.item.url ? { href: this.props.item.url } : {}}>
                    <div className="uk-card-body">
                        <div className="uk-card-badge uk-label">{this.props.item.status}</div>
                        <h3 className="uk-card-title">{this.props.item.name}</h3>
                        <p>{this.props.item.description}</p>
                    </div>
                    <div className="uk-card-footer">
                        { this.tags() }
                    </div>
                </a>
            </div>
        );
    }

}
