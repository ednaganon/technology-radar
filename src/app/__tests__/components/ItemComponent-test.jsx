import React from 'react';
import renderer from 'react-test-renderer';

import ItemComponent from '../../components/ItemComponent';
import Item from '../../class/Item';

test('ItemComponent', () => {
    const component = renderer.create(
        <ItemComponent item={new Item({
            "name":"React.js",
            "description":"React (sometimes styled React.js or ReactJS) is an open-source JavaScript library for building user interfaces.\nIt is maintained by Facebook, Instagram and a community of individual developers and corporations. According to JavaScript analytics service Libscore, React is currently being used on the websites of Netflix, Imgur, Bleacher Report, Feedly, Airbnb, SeatGeek, HelloSign, Walmart, and others.",
            "category":"language & framework",
            "status":"assess",
            "tags":"",
            "url":"https://facebook.github.io/react/"
        })} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
