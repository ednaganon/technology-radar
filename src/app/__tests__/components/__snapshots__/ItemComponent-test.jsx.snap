// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemComponent 1`] = `
<div>
  <a
    className="uk-card uk-card-default uk-card-hover"
    href="https://facebook.github.io/react/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      className="uk-card-body"
    >
      <div
        className="uk-card-badge uk-label"
      >
        assess
      </div>
      <h3
        className="uk-card-title"
      >
        React.js
      </h3>
      <p>
        React (sometimes styled React.js or ReactJS) is an open-source JavaScript library for building user interfaces.
        It is maintained by Facebook, Instagram and a community of individual developers and corporations. According to JavaScript analytics service Libscore, React is currently being used on the websites of Netflix, Imgur, Bleacher Report, Feedly, Airbnb, SeatGeek, HelloSign, Walmart, and others.
      </p>
    </div>
    <div
      className="uk-card-footer"
    >
      <span>
        <div
          className="uk-label"
        >
          
        </div>
        Â 
      </span>
    </div>
  </a>
</div>
`;
